Source/Core/TimeIntervalCollection.js

```diff
@@ -641,5 +647,376 @@ define([
+     * @returns {TimeIntervalCollection} The modified result parameter or a new instance if none was provided.
+     */
+    TimeIntervalCollection.fromJulianDateArray = function(options, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options)) {
+            throw new DeveloperError('options is required.');
+        }
+        if (!defined(options.julianDates)) {
+            throw new DeveloperError('options.iso8601Array is required.');
+        }
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new TimeIntervalCollection();
+        }
+
+        var julianDates = options.julianDates;
+        var length = julianDates.length;
+        var dataCallback = options.dataCallback;
+
+        var isStartIncluded = defaultValue(options.isStartIncluded, true);
+        var isStopIncluded = defaultValue(options.isStopIncluded, true);
+        var leadingInterval = defaultValue(options.leadingInterval, false);
+        var trailingInterval = defaultValue(options.trailingInterval, false);
+        var interval;
+
+        // Add a default interval, which will only end up being used up to first interval
+        var startIndex = 0;
+        if (leadingInterval) {
+            ++startIndex;
+            interval = new TimeInterval({
+                start: Iso8601.MINIMUM_VALUE,
+                stop: julianDates[0],
+                isStartIncluded: true,
+                isStopIncluded: !isStartIncluded
+            });
+            interval.data = defined(dataCallback) ? dataCallback(interval, result.length) : result.length;
+            result.addInterval(interval);
+        }
+
+        for (var i = 0; i < length-1; ++i) {
+            var startDate = julianDates[i];
+            var endDate = julianDates[i + 1];
+
+            interval = new TimeInterval({
+                start : startDate,
+                stop : endDate,
+                isStartIncluded : (result.length === startIndex) ? isStartIncluded : true,
+                isStopIncluded : (i === (length-2)) ? isStopIncluded : false
+            });
+            interval.data = defined(dataCallback) ? dataCallback(interval, result.length) : result.length;
+            result.addInterval(interval);
+
+            startDate = endDate;
+        }
+
+        if (trailingInterval) {
+            interval = new TimeInterval({
+                start: julianDates[length-1],
+                stop: Iso8601.MAXIMUM_VALUE,
+                isStartIncluded: !isStopIncluded,
+                isStopIncluded: true
+            });
+            interval.data = defined(dataCallback) ? dataCallback(interval, result.length) : result.length;
+            result.addInterval(interval);
+        }
+
+        return result;
+    };
+
+    var scratchGregorianDate = new GregorianDate();
+    var monthLengths = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
```

Comment:
If you notice I change February's length each iteration of the loop where its used, so it can't be frozen.